docs/cli-reference/dfx-nns.md

```diff
@@ -0,0 +1,119 @@
+
+Use the `dfx nns` subcommands to interact with the Network Nervous System.
+
+The basic syntax for running `dfx nns` commands is:
+
+``` bash
+dfx nns [subcommand] [flag]
+```
+
+Depending on the `dfx nns` subcommand you specify, additional arguments, options, and flags might apply. For reference information and examples that illustrate using `dfx nns` commands, select an appropriate command.
+
+| Command                             | Description                                                                   |
+|-------------------------------------|-------------------------------------------------------------------------------|
+| [`import`](#_dfx_nns_import)        | Adds the NNS canisters to the local dfx.json as remote canisters.             |
+| [`install`](#_dfx_nns_install)      | Deploys NNS canisters to the local dfx server                                 |
+| `help`                              | Displays usage information message for a specified subcommand.                |
+
+To view usage information for a specific subcommand, specify the subcommand and the `--help` flag. For example, to see usage information for `dfx nns install`, you can run the following command:
+
+``` bash
+$ dfx nns install --help
+```
+
+
+## dfx nns import
+
+Use the `dfx nns import` command to add the NNS canisters to the local `dfx.json`.  It also downloads the did files and sets the canister IDs of the NNS cansiters so that you can make API calls to NNS canisters.
+
+### Basic usage
+
+``` bash
+$ dfx nns import
+```
+
+### Flags
+
+You can use the following optional flags with the `dfx nns import` command.
+
+| Flag                | Description                                   |
+|---------------------|-----------------------------------------------|
+| `-h`, `--help`      | Displays usage information.                   |
+| `--network-mapping` | Renames networks when installing canister ids |
+| `-V`, `--version`   | Displays version information.                 |
+
+### Examples
+
+You can use the `dfx nns import` command to get did files and so query NNS canisters.
+
+``` bash
+$ dfx nns import
+$ dfx canister call --network ic nns-governance get_pending_proposals '()'
+```
+
+## dfx nns install
+
+Use the `dfx nns install` command to install a local NNS.  This provides local ledger and governance canisters as well as the GUI canisters Internet Identity and NNS-Dapp.
+
+### Basic usage
+The local network needs to be set up with a very specific configuration:
+```
+$ cat ~/.config/dfx/networks.json
+{
+  "local": {
+    "bind": "127.0.0.1:8080",
+    "type": "ephemeral",
+    "replica": {
+      "subnet_type": "system"
+    }
+  }
+}
+```
+
+This is because:
+
+* The NNS canisters need to run on a system subnet.
+* Some canisters are comiled to run on only very specific canister IDs and hostname/port pairs.
+
+
+In addition, the local dfx server needs to be clean:
+
+``` bash
+$ nohup dfx start --clean
```

Comment:
This is the only place I've seen `nohup`. Is it required?

Reply:
`dfx start --background` spams the screen with output even though it's meant to be running in the background whereas nohup puts stdout and stderr in a file, and also keeps running if the terminal is killed.  We could emulate the nuohup behaviour in in `--background`, although it might be better not to reinvent the wheel, use nohup and remove `--background`.  I don't mind either way, although I err on the side of reusing common, well established tools.  It's up to the sdk team to choose.  Piping the output of `--background` to some files should not be too hard.  `--background` probably inherits the parent stdout and error; changing those two lines should be enough.  But that is outside the scope of this PR.  I can make the change if wanted though.

Reply:
Possible footgun: Debug prints won't be visible with `nohup`. Also, the replica now defaults to log level `ERR`, so I think this should be fine to just use `--background`. But I don't care all that much about it, feel free to leave this as is.

Reply:
True.  The current behaviour whereby `dfx start --background` starts in the foreground is nice in that any startup errors are visible, and its those errors that are most likely to be interesting.  It's just the later spam that is really quite annoying.  Maybe those later errors could be sent to a file.  Tell the user which files are getting stdout and stderr, then background.